### ISSUE TITLE
Use a better algorithm to solve Kepler equation

### ISSUE THREAD

**Issue by abhisrkckl** [nanograv/PINT#1427]
Right now the Kepler equation is solved using the vanilla Newton-Raphson method. https://github.com/nanograv/PINT/blob/18c643a9c698120d457813f1a61e91aa8c63473d/src/pint/models/stand_alone_psr_binaries/binary_generic.py#L350

There are faster (and no less accurate) methods for doing this.
For example, Markley (1995) http://adsabs.harvard.edu/abs/1995CeMDA..63..101M

We should implement such a method instead of the Newton-Raphson method.

**Comment by dlakaplan** [comment]
This seems like a good student project

**Comment by AkashA98** [comment]
I am quite interested in this stuff... If this is not time-sensitive, I can take this up...

**Comment by scottransom** [comment]
I think this is a fine project, but we should temper our expectations!  I suspect that the current solver isn't even close to the bottleneck in any of our calculations. N-R usually converges in only a couple iterations except in high eccentricity cases, I believe.

**Comment by dlakaplan** [comment]
Yes, a more thorough analysis of where time is spent (which a few people have started but not finished) could be helpful to really focus us.

**Comment by abhisrkckl** [comment]
Here is an implementation of this
https://pyastronomy.readthedocs.io/en/latest/pyaslDoc/aslDoc/keplerOrbitAPI.html
